src/app/public/modules/tabs/tabset.component.ts

```diff
@@ -253,6 +253,7 @@ export class SkyTabsetComponent implements AfterViewInit, OnDestroy {
   public ngAfterViewInit(): void {
 
     this.initTabComponents();
+    this.listenTabComponentsStructuralChange();
```

Comment:
Instead of changing the order of operations (I'm wary of this because every change we make to tabs seems to cause someone else's implementation to break), what if we put a conditional check where the error occurs? From what I could gather, the error comes from the `SkyTabsetService`'s `unregisterTab` method when the found array index is less than zero. What if we did something like the following? (Your test should still work with this change.)

```
public unregisterTab(tabIndex: SkyTabIndex): void {
  const index = this.tabs.findIndex(tab => this.tabIndexesEqual(tab.tabIndex, tabIndex));

  // Active tab index not found.
  if (index < 0) {
    return;
  }

  // ...
}
```

Reply:
The tab is never getting registered (and won't ever because it is hitting a time when we aren't watching for changes) so I don't feel comfortable making the change in `unregisterTab` as anything where `this.tabs` is used won't ever have this tab. So all of the "active tab" code to activate things and such will break inside of the tab service.